app/src/main/java/app/gamenative/service/SteamService.kt

```
@@ -2775,42 +2789,254 @@ class SteamService : Service(), IChallengeUrlChanged {
+            try {
+                // this call errors out if run on the main thread
+                client.connect()
+
+                delay(5000)
+
+                if (!isConnected) {
+                    Timber.w("Failed to connect to Steam, marking endpoint bad and force disconnecting")
+
+                    try {
+                        client.servers.tryMark(client.currentEndpoint, PROTOCOL_TYPES, ServerQuality.BAD)
+                    } catch (e: NullPointerException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)